lib/config.py

```diff
@@ -188,47 +189,55 @@ def _setup_config_parser_options(self, subcommands):
         self.parser.add_argument('--log-size',
                                  help='Size in bytes above which the log file '
                                  'should rotate', type=int, default=2<<20)
-        self._add_subparsers(subcommands)
-
-    def _add_subparsers(self, subcommands):
-        """
-        Add subcommands to the command line parser
 
-        Args:
-            subcommands ([(str, str, [dict])]): see _setup_config_parser_options()
-        """
         subparsers = self.parser.add_subparsers(
             dest="subcommand",
             help="Available subcommands")
-
         for command, help_msg, arg_groups in subcommands:
             command_parser = subparsers.add_parser(command, help=help_msg)
             for arg_group in arg_groups:
                 for arg, options in arg_group.items():
                     command_parser.add_argument(*arg, **options)
 
-    def parse_arguments_list(self, args):
+
+    def parse_command_line(self):
         """
-        Parses the arguments provided in the argument list and returns
-            the result object.
+        Parse configuration from the command line
+
+        Returns:
+            dict: Command line options provided by user. Key is option name,
+                value is option value.
         """
+        args = sys.argv[1:]
         result = self.parser.parse_args(args)
         return vars(result)
 
+
     def parse_config_file(self, config_file_path):
         """
-        Parse the configuration file and return a dictionary containing the
-            parsed values.
+        Parse configuration from a configuration file
+
+        Args:
+            config_file_path(str): configuration file path
+
+        Returns:
+            dict: Configuration options provided by user in configuration file.
+                Key is option name, value is option value.
         """
         conf = {}
         with open(config_file_path) as stream:
             conf = yaml.safe_load(stream)
         return conf
 
+
     def parse(self):
+        """
+        Parse configuration
```

Comment:
This doesn't say much. What about:
"Parse configuration file and command line arguments"